robottelo/hosts.py

```diff
@@ -940,3 +960,18 @@ def capsule_certs_generate(self, capsule, **extra_kwargs):
         install_cmd = InstallerCommand.from_cmd_str(cmd_str=result.stdout)
         install_cmd.opts['certs-tar-file'] = f'/root/{capsule.hostname}-certs.tar'
         return install_cmd
+
+    def __enter__(self):
+        """Satellite objects can be used as a context manager to temporarily force everything
+        to use the Satellite object's hostname.
+        """
+        self.__old_hostname = settings.server.hostname
```

Comment:
What is this __old_hostname ? is that will be created by default based on the properties?  If we are planning to run destructive tests as separate threads do we required to store the existing hostname value? what is the worth

Reply:
This is for the context manager to store the current hostname so it can be reset upon exit

Reply:
Yeah you mentioned in the call that we are not going to run this as separate pytest execution in which it make sense, you might need that in running other tests